File: challenges/realm/leetcode/reverse_string.py
"""
Solution for 344. Reverse String.

https://leetcode.com/problems/reverse-string/
"""
from typing import List


class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """

        def swap(i: int, j: int) -> None:
            tmp = s[i]
            s[i] = s[j]
            s[j] = tmp

        start = 0
        end = len(s) - 1

        while start <= end:
            swap(start, end)
            start += 1
            end -= 1

File: challenges/leetcode/reverse_string.cpp
/*
 * Solution for LC#344: Reverse String
 *
 * https://leetcode.com/problems/reverse-string/
 */
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    void swap(vector<char>& s, int i, int j) {
        char tmp = s[i];
        s[i] = s[j];
        s[j] = tmp;
    }

    void reverseString(vector<char>& s) {
        int n = s.size();

        if (n == 1)
            return;

        for (int i = 0; i < n / 2; i++) {
            swap(s, i, n - 1 - i);
        }
    }
};
